Post memory cleanup

viliamDokov opened the pull request:
Moved some longe into Spotify utils instead of in the UI itself

andreitudorp reviewed (approved):
Looks good! 
Btw, I could take on the task to move the remaining methods to an util class (LocationUtil).

TPGamesNL commented:
I think it'd be better to handle the TODOs you added in this PR, before merging

TPGamesNL commented:
Also, something you could work on is removing the static resultTrack stuff in SearchFragment

viliamDokov commented:
What did I do?
1. I created a view model object which now holds the values for location, Song and Image. 
2. Location, Song and image are now wrapped in a LiveData object which basically makes them observable. This way we can more elegantly update the UI when the data changes by simply attaching an observer
3. Did some minor cleanup in PostFragment

What I didn't do:
1. Moved the location mehtods outside the PostFragment. Sadly we need to attach context (which is not hard to do) and setup a callback which is (a bit more annoying) if anyone wants the hassle go ahead!
2. Added javadoc... Yea not today

Pradyuman7 commented on app/src/main/java/com/groupseven/musicmap/MusicMap.java:
```diff
@@ -40,7 +40,7 @@ public void onCreate() {
         // Start internet check
         Intent intent = new Intent(this, InternetCheckService.class);
-        startService(intent);
+//        startForegroundService(intent);
```
Why are we not running the InternetCheckService now? Is it intentional?

Pradyuman7 replied:
Yes, I was thinking that we should probably use `startForegroundService` since we target API above 26. Iirc Android has become quite strict for background operations above API 26, and since our app can be in background the service will then crash the app.

viliamDokov replied:
Yes indeed startForegorundService crashed the app. https://pastebin.com/5N7Y7HjA. I do not know why so I commented it out. Only problem is that I forgot to mention it

viliamDokov replied:
Although this pastebin is from the foreground service. But yeah, my program also crashed from the startService

Pradyuman7 replied:
Yes, never mind my comment above. I looked into the docs and I think I found the issue (related to the reason I mentioned above), but it's kind of out of scope for this PR, so I'll create a separate PR to fix that. For now we can just leave it as `startService`.

Pradyuman7 commented on app/src/main/java/com/groupseven/musicmap/screens/main/musicmemory/create/PostFragment.java:
```diff
@@ -303,79 +292,49 @@ private String getLocationText(Location location){
                     authorID,
-                    timePosted,
-                    geoPointLocation,
+                    Calendar.getInstance().getTime(),
```
Should we try and put UTC time everywhere in the db? We can also check if this is the case currently.

viliamDokov replied:
I think it is the case. The only other place where we have timestamps in our db is in user register and it is the same there

TPGamesNL commented:
Continued in https://github.com/DBL-App-development-2023-G7/musicmap/pull/114
